be/src/storage/lake/vertical_compaction_task.cpp

```diff
@@ -192,15 +200,30 @@ Status VerticalCompactionTask::compact_column_group(bool is_key, int column_grou
             source_masks->clear();
         }
 
-        progress->update((100 * column_group_index + 100 * reader.stats().raw_rows_read / _total_num_rows) /
-                         column_group_size);
-        VLOG_EVERY_N(3, 1000) << "Tablet: " << _tablet.id() << ", compaction progress: " << progress->value();
+        _context->progress.update((100 * column_group_index + 100 * reader.stats().raw_rows_read / _total_num_rows) /
+                                  column_group_size);
+        VLOG_EVERY_N(3, 1000) << "Tablet: " << _tablet.id() << ", compaction progress: " << _context->progress.value();
     }
     RETURN_IF_ERROR(writer->flush_columns());
 
+    // add reader stats
+    auto stats = reader.stats();
+    _context->stats->reader_time_ns += reader_time_ns;
+    _context->stats->io_ns += stats.io_ns;
+    _context->stats->segment_init_ns += stats.segment_init_ns;
+    _context->stats->column_iterator_init_ns += stats.column_iterator_init_ns;
+    _context->stats->io_count_local_disk += stats.io_count_local_disk;
+    _context->stats->io_count_remote += stats.io_count_remote;
+    _context->stats->compressed_bytes_read += stats.compressed_bytes_read;
+
     if (is_key) {
         RETURN_IF_ERROR(mask_buffer->flush());
     }
+
+    auto elapsed = MonotonicNanos() - start;
```

Comment:
why need this VLOG, if it's vlog, it's likely that it will never be used

Reply:
ok, i have removed it

Reply:
have removed it